elastic/kibana
# [Fleet] Support migrating from integration type package to input type package

## Summary

Closes #137751

Minor fixes and api tests to support migrating a package from type "integration" to type "input".

**How to test:**
- download the test packages: [input-packages-upgrade.zip](https://github.com/elastic/kibana/files/9627681/input-packages-upgrade.zip)
- start a local registry:
```
docker run -p 8080:8080 -v <path_to_test_folder>/packages:/packages/test-packages -v <path_to_test_folder>/package_registry_config.yml:/package-registry/config.yml docker.elastic.co/package-registry/package-registry:main
```
- add this to your local kibana config:
```
xpack.fleet.registryUrl: http://localhost:8080
```
- This makes the custom_logs package available, with version 0.9.0 which is an integration package, and 1.0.0 which is an input type package and adds a new mandatory param "tags"
- install and create a package policy for 0.9.0 
- upgrade to 1.0.0 
- verify variables are correctly populated

### Checklist

Delete any items that are not applicable to this PR.
